modules/backend/models/EditorSetting.php

```diff
@@ -115,27 +132,57 @@ protected function makeStylesForTable($arr)
         });
     }
 
+    protected function makeFormatsForTable($arr)
+    {
+        $count = 0;
+
+        return array_build($arr, function ($key, $value) use (&$count) {
+            return [$count++, ['format_label' => $value, 'format_tag' => $key]];
+        });
+    }
+
     /**
-     * Same as getConfigured but uses special style structure.
+     * Same as getConfigured but uses a special structure for styles.
      * @return mixed
      */
     public static function getConfiguredStyles($key, $default = null)
+    {
+        return static::getConfiguredArray($key, $default, function ($key, $value) {
+            if (array_has($value, ['class_name', 'class_label'])) {
+                return [
+                    array_get($value, 'class_name'),
+                    array_get($value, 'class_label')
+                ];
+            }
+        });
+    }
+
+    /**
+     * Same as getConfigured but uses a special structure for paragraph formats.
+     * @return mixed
+     */
+    public static function getConfiguredFormats($key, $default = null)
+    {
+        return static::getConfiguredArray($key, $default, function ($key, $value) {
+            if (array_has($value, ['format_tag', 'format_label'])) {
+                return [
+                    array_get($value, 'format_tag'),
+                    array_get($value, 'format_label')
+                ];
+            }
+        });
+    }
+
+    private static function getConfiguredArray($key, $default = null, $callback = null)
```

Comment:
```suggestion
    protected static function getConfiguredArray($key, $default = null, $callback = null)
```
October pretty much never uses the private scope since everything should be extensible.